repairnator/repairnator-dockerpool/src/main/java/fr/inria/spirals/repairnator/dockerpool/AbstractPoolManager.java

```diff
@@ -93,20 +93,32 @@ public void setEngines(List<SerializerEngine> engines) {
         this.engines = engines;
     }
 
-    public RunnablePipelineContainer submitBuild(String imageId, int buildId) {
+    public TreatedBuildTracking prepareBeforeSubmitBuild(int buildId) {
         this.cleanUpOlderContainers();
-        TreatedBuildTracking treatedBuildTracking = null;
         try {
-            treatedBuildTracking = new TreatedBuildTracking(this.engines, this.runId, buildId);
+            return new TreatedBuildTracking(this.engines, this.runId, buildId);
         } catch (IOException e) {
             e.printStackTrace();
         }
+        return null;
+    }
+
+    public RunnablePipelineContainer submitBuild(String imageId, int buildId) {
+        TreatedBuildTracking treatedBuildTracking = this.prepareBeforeSubmitBuild(buildId);
         RunnablePipelineContainer runnablePipelineContainer = new RunnablePipelineContainer(this, imageId, buildId, this.dockerOutputDir, treatedBuildTracking, this.skipDelete, this.createOutputDir);
         this.submittedRunnablePipelineContainers.add(runnablePipelineContainer);
 
         return runnablePipelineContainer;
     }
 
+    public RunnablePipelineContainer submitBuild(String imageId, int buildId, int nextBuildId) {
```

Comment:
Instead of creating another method here, we should use a `InputBuildId` object as argument, don't you think?

Reply:
Way better. Done.